src/errorpage.cc

```diff
@@ -962,18 +963,13 @@ ErrorState::compileLegacyCode(Build &build)
     case 'D':
         if (!build.allowRecursion)
             p = "%D";  // if recursion is not allowed, do not convert
-#if USE_OPENSSL
-        // currently only SSL error details implemented
         else if (detail) {
-            detail->useRequest(request.getRaw());
-            const String &errDetail = detail->toString();
-            if (errDetail.size() > 0) {
-                const auto compiledDetail = compileBody(errDetail.termedBuf(), false);
-                mb.append(compiledDetail.rawContent(), compiledDetail.length());
-                do_quote = 0;
-            }
+            auto rawDetail = detail->verbose(request);
+            // TODO: Support compilation of unterminated/SBuf bodies.
+            const auto compiledDetail = compileBody(rawDetail.c_str(), false);
```

Comment:
commit 3a88ef0ea3
Personally I believe it is good idea to always allocate one more byte in SBuf just for C-string termination  if required.

Reply:
It is not the length of buffer that causes the reallocation. It is the need to cow() to prevent the terminator injection corrupting any SBuf which shares and overlaps the memory buffer.